core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -141,27 +143,36 @@ abstract class AbstractFetcherThread(name: String,
    * Builds offset for leader epoch requests for partitions that are in the truncating phase based
    * on latest epochs of the future replicas (the one that is fetching)
    */
-  private def buildLeaderEpochRequest(): ResultWithPartitions[Map[TopicPartition, EpochData]] = inLock(partitionMapLock) {
-    val partitionsWithoutEpochs = mutable.Set.empty[TopicPartition]
+  private def fetchTruncatingPartitions(): (Map[TopicPartition, EpochData], Set[TopicPartition]) = inLock(partitionMapLock) {
     val partitionsWithEpochs = mutable.Map.empty[TopicPartition, EpochData]
+    val partitionsWithoutEpochs = mutable.Set.empty[TopicPartition]
 
     partitionStates.stream().forEach(new Consumer[PartitionStates.PartitionState[PartitionFetchState]] {
```

Comment:
nit: IntelliJ suggests to convert this to a `Single Abstract Method`, effectivelly removing the `new Consumer[PartitionStates.PartitionState[PartitionFetchState]]` and ` override def accept(state: PartitionStates.PartitionState[PartitionFetchState]): Unit = {` boilerplate

Reply:
I think that only works with scala 2.12.